src/ext-test/java/org/opentripplanner/ext/realtimeresolver/RealtimeResolverTest.java

```diff
@@ -0,0 +1,481 @@
+      return null;
+    }
+
+    @Override
+    public Timetable getTimetableForTripPattern(TripPattern tripPattern, LocalDate serviceDate) {
+      return null;
+    }
+
+    @Override
+    public TripPattern getRealtimeAddedTripPattern(FeedScopedId tripId, LocalDate serviceDate) {
+      return null;
+    }
+
+    @Override
+    public boolean hasRealtimeAddedTripPatterns() {
+      return false;
+    }
+
+    @Override
+    public TripOnServiceDate getTripOnServiceDateForTripAndDay(
+      TripIdAndServiceDate tripIdAndServiceDate
+    ) {
+      return null;
+    }
+
+    @Override
+    public TripOnServiceDate getTripOnServiceDateById(FeedScopedId datedServiceJourneyId) {
+      return null;
+    }
+
+    @Override
+    public Collection<TripOnServiceDate> getAllTripOnServiceDates() {
+      return null;
+    }
+
+    @Override
+    public Set<TransitMode> getTransitModes() {
+      return null;
+    }
+
+    @Override
+    public Collection<PathTransfer> getTransfersByStop(StopLocation stop) {
+      return null;
+    }
+
+    @Override
+    public TransitLayer getTransitLayer() {
+      return null;
+    }
+
+    @Override
+    public TransitLayer getRealtimeTransitLayer() {
+      return null;
+    }
+
+    @Override
+    public CalendarService getCalendarService() {
+      return null;
+    }
+
+    @Override
+    public ZoneId getTimeZone() {
+      return null;
+    }
+
+    @Override
+    public TransitAlertService getTransitAlertService() {
+      return null;
+    }
+
+    @Override
+    public FlexIndex getFlexIndex() {
+      return null;
+    }
+
+    @Override
+    public ZonedDateTime getTransitServiceEnds() {
+      return null;
+    }
+
+    @Override
+    public ZonedDateTime getTransitServiceStarts() {
+      return null;
+    }
+
+    @Override
+    public Optional<Coordinate> getCenter() {
+      return Optional.empty();
+    }
+
+    @Override
+    public TransferService getTransferService() {
+      return null;
+    }
+
+    @Override
+    public boolean transitFeedCovers(Instant dateTime) {
+      return false;
+    }
+
+    @Override
+    public Collection<RegularStop> findRegularStop(Envelope envelope) {
+      return null;
+    }
+
+    @Override
+    public GraphUpdaterStatus getUpdaterStatus() {
+      return null;
+    }
+  }
+
+  @Test
+  void testPopulateLegsWithRealtime() {
+    var legs = new ArrayList<Leg>();
+    legs.add(new MockLeg());
+    legs.add(new MockLeg());
+    legs.add(new MockLeg());
+
+    var itineraries = new ArrayList<Itinerary>();
+    itineraries.add(new Itinerary(legs));
+    itineraries.add(new Itinerary(legs));
+    itineraries.add(new Itinerary(legs));
+
+    RealtimeResolver.populateLegsWithRealtime(itineraries, new MockTransitService());
+
+    assertEquals(3, itineraries.size());
+
+    itineraries.forEach(it -> {
+      var lgs = it.getLegs();
+      assertEquals(3, lgs.size());
+      lgs.forEach(l -> assertEquals(MOCK_LEG, l));
+    });
+  }
```

Comment:
I don't really think that the test brings any value in its current state. Basically you are just checking that `populateLegsWithRealtime` does not do anything with `itineraries`

Reply:
It verifies that we are using leg reference to find potential realtime version of the leg. Not sure how otherwise we could test it. It will be quite complicated to mock whole transit model with realtime layer on top.

Reply:
Yes, I've added it now.